rskj-core/src/main/java/co/rsk/util/HexUtils.java

```diff
@@ -0,0 +1,336 @@
+import org.bouncycastle.util.encoders.Hex;
+import org.ethereum.util.ByteUtil;
+
+import static org.ethereum.rpc.exception.RskJsonRpcRequestException.invalidParamError;
+
+import java.math.BigInteger;
+import java.nio.charset.StandardCharsets;
+import java.util.regex.Pattern;
+
+/**
+ * Hex utils
+ */
+public class HexUtils {
+	
+    private static final Pattern LEADING_ZEROS_PATTERN = Pattern.compile("0x(0)+");
+
+    private static final String HEX_PREFIX = "0x";
+
+    private static final String ZERO_STR = "0";
+
+    private static final byte[] HEX_PREFIX_BYTE_ARRAY = HEX_PREFIX.getBytes();
+
+    private static final byte[] ZERO_BYTE_ARRAY = "0".getBytes();
+
+    private static final String INCORRECT_HEX_SYNTAX = "Incorrect hex syntax";
+    private static final String INPUT_CANT_BE_NULL = "input cant be null";
+    private static final String PREFIX_EXPECTED = "Invalid hex number, expected 0x prefix";
+
+    private HexUtils() {
+        throw new IllegalAccessError("Utility class");
+    }
+
+    /**
+     * If 0x prefix is present Convert hex encoded string to decoded BigInteger
+     * Else converts number in String representation to BigInteger
+     */
+    public static BigInteger stringNumberAsBigInt(final String input) {
+        if (hasHexPrefix(input)) {
+            return HexUtils.stringHexToBigInteger(input);
+        } else {
+            return new BigInteger(input);
+        }
+    }
+
+    /**
+     * Convert hex encoded string to decoded BigInteger
+     */
+    public static BigInteger stringHexToBigInteger(final String input) {
+        if(!hasHexPrefix(input)) {
+            throw new NumberFormatException(PREFIX_EXPECTED);
+        }
+        String hexa = input.substring(2);
+        return new BigInteger(hexa, 16);
+    }
+
+    /**
+     * Convert hex encoded string to decoded byte array
+     */
+    public static byte[] stringHexToByteArray(final String param) {
+
+        if (param == null) {
+            throw invalidParamError(INPUT_CANT_BE_NULL);
+        }
+
+        String result = removeHexPrefix(param);
+
+        if (result.length() % 2 != 0) {
+            result = ZERO_STR + result;
+        }
+        return Hex.decode(result);
+    }
+
+    /**
+     * String to byte array 
+     */
+    public static byte[] stringToByteArray(String input) {
+        return input.getBytes(StandardCharsets.UTF_8);
+    }
+
+    /**
+     * Converts a byte array to a string according to ethereum json-rpc specifications
+     */
+    public static String toJsonHex(final byte[] param) {
+        String result = toUnformattedJsonHex(param);
+
+        if (HEX_PREFIX.equals(result)) {
+            return "0x00";
+        }
+
+        return result;
+    }
+
+    /**
+     * Convert from Coin to its BigInteger value in String representation or empty if the parameter is null
+     */
+    public static String toJsonHex(final Coin coin) {
+        return coin != null ? coin.asBigInteger().toString() : "" ;
```

Comment:
I don't think this is up to spec either, should be "0x0", right?

Reply:
This, was already present in TypeConverter